Libplanet.Action/State/CurrencyAccount.cs

```diff
@@ -0,0 +1,400 @@
+using System;
+using System.Numerics;
+using Bencodex.Types;
+using Libplanet.Crypto;
+using Libplanet.Store.Trie;
+using Libplanet.Types.Assets;
+using Libplanet.Types.Blocks;
+
+namespace Libplanet.Action.State
+{
+    /// <summary>
+    /// A special "account" for managing <see cref="Currency"/> starting with
+    /// <see cref="BlockMetadata.CurrencyAccountProtocolVersion"/>.
+    /// </summary>
+    public class CurrencyAccount
```

Comment:
I think there is no reason for this class to be inherited, if it is, this class should be sealed.

Reply:
Applied 🙂